jwst/resample/resample.py

```diff
@@ -1,24 +1,700 @@
+            # build ModelLibrary as a list of in-memory models
+            return ModelLibrary(output_models, on_disk=False)
+        else:
+            # build ModelLibrary as an association from the output files
+            # this saves memory if there are multiple groups
+            asn = asn_from_list(output_models, product_name='outlier_i2d')
+            asn_dict = json.loads(asn.dump()[1])  # serializes the asn and converts to dict
+            return ModelLibrary(asn_dict, on_disk=True)
+
+    def resample_many_to_one(self):
+        """Resample and coadd many inputs to a single output.
+
+        Used for stage 3 resampling
+        """
+        # if self.output_jwst_model is not None:
+        #     self.reset_arrays(reset_output=True)
+
+        log.info("Resampling science and variance data")
+
+        with self.input_models:
+            for model in self.input_models:
+                self.add_model(model)
+                self.input_models.shelve(model)
+
+        self.finalize()
+        self.output_jwst_model.meta.filename = self.output_filename
+        copy_asn_info_from_library(self.input_models, self.output_jwst_model)
+
+        return self.output_jwst_model
+
+    @staticmethod
+    def update_fits_wcsinfo(model):
+        """
+        Update FITS WCS keywords of the resampled image.
+        """
+        # Delete any SIP-related keywords first
+        pattern = r"^(cd[12]_[12]|[ab]p?_\d_\d|[ab]p?_order)$"
+        regex = re.compile(pattern)
+
+        keys = list(model.meta.wcsinfo.instance.keys())
+        for key in keys:
+            if regex.match(key):
+                del model.meta.wcsinfo.instance[key]
+
+        # Write new PC-matrix-based WCS based on GWCS model
+        transform = model.meta.wcs.forward_transform
+        model.meta.wcsinfo.crpix1 = -transform[0].offset.value + 1
+        model.meta.wcsinfo.crpix2 = -transform[1].offset.value + 1
+        model.meta.wcsinfo.cdelt1 = transform[3].factor.value
+        model.meta.wcsinfo.cdelt2 = transform[4].factor.value
+        model.meta.wcsinfo.ra_ref = transform[6].lon.value
+        model.meta.wcsinfo.dec_ref = transform[6].lat.value
+        model.meta.wcsinfo.crval1 = model.meta.wcsinfo.ra_ref
+        model.meta.wcsinfo.crval2 = model.meta.wcsinfo.dec_ref
+        model.meta.wcsinfo.pc1_1 = transform[2].matrix.value[0][0]
+        model.meta.wcsinfo.pc1_2 = transform[2].matrix.value[0][1]
+        model.meta.wcsinfo.pc2_1 = transform[2].matrix.value[1][0]
+        model.meta.wcsinfo.pc2_2 = transform[2].matrix.value[1][1]
+        model.meta.wcsinfo.ctype1 = "RA---TAN"
+        model.meta.wcsinfo.ctype2 = "DEC--TAN"
+
+        # Remove no longer relevant WCS keywords
+        rm_keys = ['v2_ref', 'v3_ref', 'ra_ref', 'dec_ref', 'roll_ref',
+                'v3yangle', 'vparity']
+        for key in rm_keys:
+            if key in model.meta.wcsinfo.instance:
+                del model.meta.wcsinfo.instance[key]
 
 
 class ResampleData:
```

Comment:
I'm a bit confused about why the majority of ResampleData hasn't been removed, as it appears that functionality is mostly what got moved to stcal.  What would it take to allow large portions (or all) of this class to be removed?